.gitlab-ci.yml

```diff
@@ -12,53 +10,59 @@ stages:
   - test
   - build
   - image
-# before_script:
-#   - mkdir -p /go/src/gitlab.com/pantomath-io /go/src/_/builds
-#   - cp -r $CI_PROJECT_DIR /go/src/gitlab.com/pantomath-io/pantomath
-#   - ln -s /go/src/gitlab.com/pantomath-io /go/src/_/builds/pantomath-io
-#   - make dep
 
-unit_tests:
+.golang_based:
   tags:
     - docker
+  image: golang:1.17
+
+unit_tests:
+  extends:
+    - .golang_based
   stage: test
   script:
     - make test
 
 # race_detector:
+#   extends:
+#     - .golang_based
 #   stage: test
 #   script:
 #     - make race
 
 # memory_sanitizer:
+#   extends:
+#     - .golang_based
 #   stage: test
 #   script:
 #     - make msan
 
 code_coverage:
-  tags:
-    - docker
+  extends:
+    - .golang_based
   stage: test
   script:
     - make cov
 
 code_coverage_report:
-  tags:
-    - docker
+  extends:
+    - .golang_based
   stage: test
   script:
     - make covhtml
   only:
   - master
 
 # lint_code:
+#   extends:
+#     - .golang_based
 #   stage: test
 #   script:
 #     - make lint
 
 build:
```

Comment:
```suggestion
build_and_publish:
```

Reply:
You do not plan to add a deploy step ?

Reply:
No, it's in a different job but I rename this job `build-publish-docker`